test/Microsoft.Net.Http.Headers.Tests/MediaTypeHeaderValueTest.cs

```diff
@@ -681,6 +746,74 @@ public void IsSubsetOf_NegativeCases(string mediaType1, string mediaType2)
             Assert.False(isSubset);
         }
 
+        [Theory]
+        [InlineData("application/entity+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/entity+json")]
+        [InlineData("application/*+json", "application/*+json")]
+        [InlineData("application/*", "application/*+JSON")]
+        [InlineData("application/vnd.github+json", "application/vnd.github+json")]
+        [InlineData("application/*", "application/entity+JSON")]
+        [InlineData("*/*", "application/entity+json")]
+        public void IsSubsetOfWithSuffixes_PositiveCases(string set, string subset)
+        {
+            // Arrange
+            var setMediaType = MediaTypeHeaderValue.Parse(set);
+            var subSetMediaType = MediaTypeHeaderValue.Parse(subset);
+
+            // Act
+            var result = subSetMediaType.IsSubsetOf(setMediaType);
+
+            // Assert
+            Assert.True(result);
+        }
+
+        [Theory]
+        [InlineData("application/entity+json", "application/entity+txt")]
+        [InlineData("application/entity+json", "application/entity.v2+json")]
+        [InlineData("application/*+json", "application/entity+txt")]
+        [InlineData("application/*+*", "application/json")]
+        [InlineData("application/entity+*", "application/entity+json")] // We don't allow suffixes to be wildcards
+        [InlineData("application/*+*", "application/entity+json")] // We don't allow suffixes to be wildcards
+        public void IsSubSetOfWithSuffixes_NegativeCases(string set, string subset)
+        {
+            // Arrange
+            var setMediaType = MediaTypeHeaderValue.Parse(set);
+            var subSetMediaType = MediaTypeHeaderValue.Parse(subset);
+
+            // Act
+            var result = subSetMediaType.IsSubsetOf(setMediaType);
+
+            // Assert
+            Assert.False(result);
+        }
+
+        public static TheoryData<string, List<StringSegment>> MediaTypesWithFacets =>
+                 new TheoryData<string, List<StringSegment>>
+                 {
+                     { "application/vdn.github",
+                         new List<StringSegment>(){ new StringSegment("vdn"), new StringSegment("github") } },
+                     { "application/vdn.github+json",
+                         new List<StringSegment>(){ new StringSegment("vdn"), new StringSegment("github") } },
```

Comment:
There's an implicit converter from string to StringSegment, you don't need to say `new StringSegment` here do you?